include/nbl/asset/IRayTracingPipeline.h

```diff
@@ -0,0 +1,163 @@
+#ifndef _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+#define _NBL_ASSET_I_RAY_TRACING_PIPELINE_H_INCLUDED_
+
+
+#include "nbl/asset/IShader.h"
+#include "nbl/asset/RasterizationStates.h"
+#include "nbl/asset/IPipeline.h"
+#include "nbl/asset/IRenderpass.h"
+
+#include <span>
+
+
+namespace nbl::asset
+{
+  struct SShaderGroupsParams
+  {
+    constexpr static inline uint32_t ShaderUnused = 0xffFFffFFu;
+
+    struct SGeneralGroup
+    {
+      uint32_t shaderIndex = ShaderUnused;
+    };
+
+    struct SHitGroup
+    {
+      uint32_t closestHitShaderIndex = ShaderUnused;
+      uint32_t anyHitShaderIndex = ShaderUnused;
+      uint32_t intersectionShaderIndex = ShaderUnused;
+    };
+
+    SGeneralGroup raygenGroup;
+    core::vector<SGeneralGroup> missGroups;
+    core::vector<SHitGroup> hitGroups;
+    core::vector<SGeneralGroup> callableGroups;
+
+    inline uint32_t getShaderGroupCount() const
+    {
+      return 1 + hitGroups.size() + missGroups.size() + callableGroups.size();
+    }
+
+  };
+  using SGeneralShaderGroup = SShaderGroupsParams::SGeneralGroup;
+  using SHitShaderGroup = SShaderGroupsParams::SHitGroup;
+  ;
+
+  class IRayTracingPipelineBase : public virtual core::IReferenceCounted
+  {
+  public:
+
+    struct SCachedCreationParams final
+    {
+      SShaderGroupsParams shaderGroups;
```

Comment:
the groups can't be cached verbatim, need to be copied into local storage (because they must use spans